nir/src/main/scala/scala/scalanative/nir/Ops.scala

```diff
@@ -110,37 +111,95 @@ sealed abstract class Op {
 }
 object Op {
   // low-level
-  final case class Call(ty: Type, ptr: Val, args: Seq[Val])         extends Op
-  final case class Load(ty: Type, ptr: Val)                         extends Op
-  final case class Store(ty: Type, ptr: Val, value: Val)            extends Op
-  final case class Elem(ty: Type, ptr: Val, indexes: Seq[Val])      extends Op
-  final case class Extract(aggr: Val, indexes: Seq[Int])            extends Op
-  final case class Insert(aggr: Val, value: Val, indexes: Seq[Int]) extends Op
-  final case class Stackalloc(ty: Type, n: Val)                     extends Op
-  final case class Bin(bin: nir.Bin, ty: Type, l: Val, r: Val)      extends Op
-  final case class Comp(comp: nir.Comp, ty: Type, l: Val, r: Val)   extends Op
-  final case class Conv(conv: nir.Conv, ty: Type, value: Val)       extends Op
+  final case class Call(ty: Type, ptr: Val, args: Seq[Val]) extends Op {
+    override lazy val hashCode: Int = MurmurHash3.productHash(this)
```

Comment:
How about defining cached hashCode in `Op` itself?
We can use something like: 
```scala
sealed abstract class Op {
  self: Product =>
  override lazy val hashCode = MurmurHash3.productHash(self)
}
```
Each instance of `Op` is a case class so it would always be possible to apply. 
It definitely would make it easier to maintain as we'll never forget to override hashCode.